samples/sample_common/src/vaapi_device.cpp

```diff
@@ -95,7 +95,9 @@ mfxStatus CVAAPIDeviceX11::Init(mfxHDL hWindow, mfxU16 nViews, mfxU32 nAdapterNu
     }
     m_xcbconn = x11xcblib.XGetXCBConnection(m_dpy);
 
-    m_dri_fd = open("/dev/dri/card0", O_RDWR);
+    // it's enough to pass render node, because we only request
+    // information from kernel via m_dri_fd
+    m_dri_fd = open("/dev/dri/renderD128", O_RDWR);
```

Comment:
this looks ok, I think.

However, the rest of the function does not make sense to me. For example, on line 67 we call XCreateSimpleWindow, thus we should already have X display opened, but we open it on the line 91 actually. Just grepping the code:
```
samples/sample_common/src/vaapi_device.cpp:91:    m_dpy = x11lib.XOpenDisplay(NULL);
samples/sample_common/src/vaapi_utils_x11.cpp:49:            m_display = m_x11lib.XOpenDisplay(currentDisplay);
samples/sample_common/src/vaapi_utils_x11.cpp:51:            m_display = m_x11lib.XOpenDisplay(VAAPI_X_DEFAULT_DISPLAY);
samples/sample_common/src/vaapi_utils_x11.cpp:136:            m_display = m_x11lib.XOpenDisplay(currentDisplay);
samples/sample_common/src/vaapi_utils_x11.cpp:138:            m_display = m_x11lib.XOpenDisplay(VAAPI_X_DEFAULT_DISPLAY);
```
Do we really need all 3 functions?

Reply:
It seems CVAAPIDeviceX11 can use connection from m_X11LibVA it includes.